src/IK/src/CoMTask.cpp

```diff
@@ -0,0 +1,153 @@
+    {
+        log()->error("[CoMTask::setVariableHandler] The size of the robot velocity variable is "
+                     "different from the one expected. Expected size: {}. Given size: {}.",
+                     m_kinDyn->getNrOfDegreesOfFreedom() + m_spatialVelocitySize,
+                     m_robotVelocityVariable.size);
+        return false;
+    }
+
+    // resize the matrices
+    m_A.resize(m_linearVelocitySize, variablesHandler.getNumberOfVariables());
+    m_A.setZero();
+    m_b.resize(m_linearVelocitySize);
+
+    return true;
+}
+
+bool CoMTask::initialize(std::weak_ptr<ParametersHandler::IParametersHandler> paramHandler)
+{
+    constexpr auto errorPrefix = "[CoMTask::initialize]";
+
+    m_description = "CoMTask Optimal Control Element";
+
+    if (m_kinDyn == nullptr || !m_kinDyn->isValid())
+    {
+        log()->error("{} [{}] KinDynComputations object is not valid.",
+                     errorPrefix,
+                     m_description);
+
+        return false;
+    }
+
+    if (m_kinDyn->getFrameVelocityRepresentation()
+        != iDynTree::FrameVelocityRepresentation::MIXED_REPRESENTATION)
+    {
+        log()->error("{} [{}] task supports only quantities expressed in MIXED "
+                     "representation. Please provide a KinDynComputations with Frame velocity "
+                     "representation set to MIXED_REPRESENTATION.",
+                     errorPrefix,
+                     m_description);
+        return false;
+    }
+
+    auto ptr = paramHandler.lock();
+    if (ptr == nullptr)
+    {
+        log()->error("{} [{}] The parameter handler is not valid.",
+                     errorPrefix,
+                     m_description);
+        return false;
+    }
+
+    std::string robotVelocityVariableName;
+    if (!ptr->getParameter("robot_velocity_variable_name", m_robotVelocityVariable.name))
+    {
+        log()->error("{} [{}] while retrieving the robot velocity variable.",
+                     errorPrefix,
+                     m_description);
+        return false;
+    }
+
+    // set the gains for the controllers
+    double kpLinear;
+    if (!ptr->getParameter("kp_linear", kpLinear))
+    {
+        log()->error("{} [{}] to get the proportional linear gain.",
+                     errorPrefix,
+                     m_description);
+        return false;
+    }
+
+
+    m_R3Controller.setGains(kpLinear);
+
+    m_isInitialized = true;
+
+    return true;
+}
+
+bool CoMTask::update()
+{
+    using namespace BipedalLocomotion::Conversions;
+    using namespace iDynTree;
+
+    // set the state
+    m_R3Controller.setState(toEigen(m_kinDyn->getCenterOfMassPosition()));
+
+    // update the controller
+    m_R3Controller.computeControlLaw();
+
+    m_b = m_R3Controller.getControl().coeffs();
+
+    // Workaround because matrix view is not compatible with Eigen::Ref
+    // https://github.com/robotology/idyntree/issues/797
```

Comment:
I think this comment is no more necessary, right?

Reply:
Done in e60156cfd83505d4493b73eab031819a08169c15